pep-0581.rst

```diff
@@ -0,0 +1,310 @@
+Status: Draft
+Type: Process
+Content-Type: text/x-rst
+Created: 20-Jun-2018
+
+
+Abstract
+========
+
+This PEP outlines the steps required to migrate Python's issue tracker
+from Roundup to using GitHub Issues.
+
+
+Rationale
+=========
+
+CPython's development moved to GitHub on February 2017. All other projects within
+The PSF's organization are hosted on GitHub and are using GitHub issues.
+CPython is still using Roundup as the issue tracker in bugs.python.org (bpo).
+
+Why GitHub
+----------
+
+It has a lot of nice features readily available out of the box, that are not
+currently available in Roundup / bpo.
+
+- APIs we can use to build integrations and automations.
+
+- Ability to attach screenshots and files in pull requests and issues.
+
+- Administrators and core developers can edit issues comments and pull requests.
+
+- Ability to reply to issues and pull request conversations via email.
+
+- It supports two factor authentication.
+
+- It supports markdown and emoji.
+
+- Core developers don't have to maintain it, giving us more time to focus on
+  the development of Python.
+
+- We can automatically close issues when a PR has been merged [#close-github-issue]_.
+
+- With more than 28 million users, an open source contributor is more likely
+  to already have an account, and familiar with GitHub interface, making it
+  easier to start contributing.
+
+- Email notifications contain metadata [#email-meta]_, integrated with GMail, and
+  allows you to systematically filter emails.
+
+- Email address can remain hidden by choice.
+
+
+Issues with bpo / Roundup
+-------------------------
+
+- It is not actively maintained.
+
+- Less than five people maintain bpo. Some of them are core developers.
+
+- It is in mercurial. Without any CI available, it puts heavy burden on existing
+  (few) maintainers in terms of reviewing, testing, and applying patches.
+
+- At its current state, it is not equipped to accept lots of contributions from
+  people who aren't already familiar with the code base.
+
+- The upstream roundup is in mercurial. If bpo does move to GitHub, it will become
+  difficult to update patches from upstream.
+
+- The user interface needs update and redesign. It will require UX/UI research
+  to keep it up to date with current web standards.
+
+- Email address is exposed. There is no choice to mask it.
+
+- There is no REST API available. There is an open issue in Roundup for adding
+  REST API  [#roundup-rest-api]_. Last activity was in 2016.
+
+- I personally don't want to receive email each time someone added themselves as "nosy"
+  An issue has been filed in upstream roundup since 2012 with little traction [#nosy-list]_.
+
+- Creating an account has been a hassle. We've had reports when people have
+  trouble creating accounts or logging in.
+
+Why not GitLab
+--------------
+
+Had we migrated to GitLab instead of GitHub in 2017, this PEP would have been
+titled "Using GitLab Issues for CPython".
+
```

Comment:
You should mention somewhere "commercial" or "vendor lock-in" and explain why you consider that it's not a blocker issue. I'm sure that the question will arise.

Reply:
The repo exports that Ernest (has set up? is setting up?) are definitely worth mentioning.

Reply:
Ah, they're already mentioned later on in a slightly different context. Still, that's the key point for addressing any lock-in concerns: there are PSF hosted data backups, and the only thing that gets potentially tricky is tracking identity following a future migration away from GitHub, and the CLA signing process can cover that.